File: movieIos/MovieCatalogue/App/Utils/FavoriteState.swift
//
//  FavoriteState.swift
//  MovieCatalogue
//
//  Created by Uwais Alqadri on 30/09/21.
//  Copyright © 2021 Uwais Alqadri. All rights reserved.
//

import Foundation

enum FavoriteState {
  case favMovieFound(state: Bool)
  case addFavorite
  case removeFavorite
}

File: movieAndroid/src/main/java/com/uwaisalqadri/moviecatalogue/android/utils/FavoriteState.kt
package com.uwaisalqadri.moviecatalogue.android.utils

import com.uwaisalqadri.moviecatalogue.domain.model.Movie

/**
 * Created by Uwais Alqadri on September 29, 2021
 */
sealed class FavoriteState
data class FavMovieDataFound(val movie: List<Movie>) : FavoriteState()
object AddFavorite : FavoriteState()
object RemoveFavorite : FavoriteState()